src/lib/free-queue/test/free-queue-sab.test.js

```diff
@@ -0,0 +1,113 @@
+import { expect } from 'https://cdnjs.cloudflare.com/ajax/libs/chai/5.1.1/chai.js';
+
+import { FreeQueueSAB } from '../free-queue-sab.js';
+
+
+describe('FreeQueueSAB', function() {
+    const bufferLength = 1024;
+    const channelCount = 2;
+    let queue;
+
+    beforeEach(() => {
+        queue = new FreeQueueSAB(bufferLength, channelCount);
+    });
+
+    describe('Constructor', function() {
+        it('should initialize states correctly', function() {
+            expect(queue.states).to.be.an.instanceof(Uint32Array);
+            expect(queue.states.length).to.equal(2);
+        });
+
+        it('should initialize buffer length correctly', function() {
+            expect(queue.getBufferLength()).to.equal(bufferLength);
+        });
+
+        it('should initialize channel data correctly', function() {
+            expect(queue.channelData).to.be.an('array');
+            expect(queue.channelData.length).to.equal(channelCount);
+            queue.channelData.forEach(channel => {
+                expect(channel).to.be.an.instanceof(Float32Array);
+                expect(channel.length).to.equal(bufferLength + 1);
+            });
+        });
+    });
+
+    describe('push', function() {
+        it('should push data into the queue', function() {
+            const input = [new Float32Array([1, 2, 3]), new Float32Array([4, 5, 6])];
+            const result = queue.push(input, 3);
+            expect(result).to.be.true;
+            expect(queue.getAvailableSamples()).to.equal(3);
+        });
+
+        it('should fail to push data when buffer is full', function() {
+            const input = [new Float32Array(queue.getBufferLength()), new Float32Array(queue.getBufferLength())];
+            
+            let result = queue.push(input, queue.getBufferLength());
+            console.log("Push result when buffer is full:", result);
+            expect(result).to.be.true;
```

Comment:
Instead of logging to the console here, let's use the message feature of the `expect()` function:
`expect(result, 'Push result when buffer is full').to.be.true;`